infra/ballista/build/lambda-handler.py

```diff
@@ -0,0 +1,80 @@
+import os
+import subprocess
+import logging
+import sys
+import shutil
+import time
+
+
+logging.getLogger().setLevel(logging.DEBUG)
+__stdout__ = sys.stdout
+
+IS_COLD_START = True
+
+def handler(event, context):
+    """An AWS Lambda handler that runs the provided command with bash and returns the standard output"""
+    shutil.rmtree("/tmp", ignore_errors=True)
+    start = time.time()
+    global IS_COLD_START
+    is_cold_start = IS_COLD_START
+    IS_COLD_START = False
+    # start scheduler
+    process_s = subprocess.Popen(["/opt/ballista/ballista-scheduler"],
+                                stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    logging.debug("scheduler starts")
+    # wait till the scheduler is up and running
+    return_code=1
+    while return_code:
+        res = subprocess.run(["nc", "-z", "localhost", "50050"], capture_output=True)
+        return_code = res.returncode
+        logging.debug(res)
+        
+    process_e = subprocess.Popen(["/opt/ballista/ballista-executor", "-c" ,"4"],
+                                 stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    logging.debug("executor.starts")
+    return_code=1
+    while return_code:
+        res = subprocess.run(["nc", "-z", "localhost", "50051"], capture_output=True)
+        return_code = res.returncode
+        logging.debug(res)
+    # input parameters
+    logging.debug("event: %s", event)
+    çomm_file_path = '/tmp/commands.sql'
+    with open(çomm_file_path,'w') as file:
+        file.write(f"""CREATE EXTERNAL TABLE trips STORED AS PARQUET LOCATION 's3://{os.getenv("DATA_BUCKET_NAME")}/nyc-taxi/2019/01/';
+SELECT payment_type, SUM(trip_distance) FROM trips GROUP BY payment_type;""")
+    command = ['/opt/ballista/ballista-cli', '--host', 'localhost', '--port', '50050', '-f', f'{çomm_file_path}', '--format', 'csv']
```

Comment:
1) I think it's better to avoid strange characters in codes, most people won't know how to reproduce them on their keyboards (`ç`)
2) have you tried piping in the data and reading from `/dev/stdin`? not 100% sure it will work but it would avoid writing this tmp thing to the file system
3) can you refactor the actual query to source it from the arguments of the handler as it is done for the other engines?

Reply:
1. I didn't see it. now is fixed
2. the file is mandatory for execute several commands (then exit) on ballista-cli [see doc](https://github.com/apache/arrow-ballista/tree/master/ballista-cli). changing this means refactoring the whole thing to have the cli in the backgroud running and then trying to communicate with it maybe using `subprocess.communicate`? 
3. this only make sense if we change the approach like commented in 2.

Reply:
2. I don't think you need to calll `communicate`, just writing to `stdin` should be enough
3. this is by no means related to 2. The query string could provided as argument to the handler, no matter how you use it internally

Reply:
2. I tested just with stdin and I cannot get it to work (it ignore the select query, seems to be a sync issue on my approach)
    It doesn't work either with communicate (communicate closes the stdin when its done writing data)
3. I disagree to a certain level. passing a string in one function or the other is not fundamentally different. to take the most if it we should be able to send continuously commands and this implies refactor.
I change the implementation to be in line with the other engines 😺